server/src/collection.rs

```diff
@@ -156,4 +152,26 @@ impl Collection {
     pub async fn alias(&self) -> String {
         self.alias.lock().await.clone()
     }
+
+    pub async fn dispatch_items(&self, object_server: &zbus::ObjectServer) -> Result<(), Error> {
+        let keyring_items = self.keyring.items().await;
+        let mut n_items: u32 = 1;
+
+        for keyring_item in keyring_items {
+            let item = item::Item::new(
+                keyring_item.map_err(Error::InvalidItem)?,
+                self.is_locked().await,
+                Arc::clone(&self.manager),
+                self.path.clone(),
+                n_items,
+            );
+            *self.item_index.write().await = n_items;
```

Comment:
I already said two times before, this should be done at the end of the for iteration. There is 0 reason to update the value every time, as this method happens at the start of the service, so no one would be creating new items until it us up and ready.

Reply:
Ah! my bad. Actually last [time](https://github.com/bilelmoussaoui/oo7/compare/5b69092873f820f3018ccb8f9f84a3fd5cde8faf..35458ea6ba4c781797652fc33be86f0110492971) it was correctly positioned.